test/test-methods/test_factory.cpp

```diff
@@ -288,4 +288,38 @@ void CPUMiner::Solve(Block& b) {
     b.SetTime(final_time.load());
     b.CalculateHash();
     b.CalculateOptimalEncodingSize();
+    return true;
+}
+
+CPUMiner::CPUMiner(size_t nThreads) : Miner() {
+    solverPool_.SetThreadSize(nThreads);
+}
+
+bool CPUMiner::Start() {
+    bool flag = false;
+    if (enabled_.compare_exchange_strong(flag, true)) {
+        solverPool_.Start();
+        spdlog::info("Miner started.");
+        return true;
+    }
+
+    return false;
+}
+
+bool CPUMiner::Stop() {
+    bool flag = true;
+    if (enabled_.compare_exchange_strong(flag, false)) {
+        spdlog::info("Miner stopping...");
+        if (runner_.joinable()) {
+            runner_.join();
+        }
+        if (inspector_.joinable()) {
+            inspector_.join();
+        }
+        solverPool_.Stop();
+        spdlog::info("Miner stopped.");
+        return true;
+    }
+
+    return false;
```

Comment:
```suggestion
bool CPUMiner::Stop() {
    if (base::Stop()) {
         solverPool_.Stop();
         return true;
    }

     return false;
```


and remove `spdlog::info(“Miner stopped.”)` from `Miner::Stop`.